### DISCUSSION TITLE
Extending the MyST syntax

### DISCUSSION THREAD

**Discussion by AdrianVollmer** [executablebooks/MyST-Parser#788] [Q&A] [answered]
Hi,

I was wondering if it was at all possible to extend the syntax, and if so, how to go about it.

My end goal would be a new feature which you could think of as sort of a "named thematic break":

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur a porta arcu, quis finibus sem.

===label===

Etiam blandit pellentesque lacus at ullamcorper.
```

where `label` can be an arbitrary alphanumeric string.

From reading the code it seems that there was no intention of extending the syntax. I'd have to first extend Markdown-It-Py, but then I'd have no way of enabling this new extension in MyST.

Please correct me if I'm wrong. If not, is this something you'd be willing to consider? If no, does anyone have suggestions of a sensible workaround?

Thanks in advance!

**Comment by chrisjsewell** [comment, marked as answer]
Heya, no it is not currently possible to arbitrarily extend the syntax, see https://github.com/executablebooks/MyST-Parser/issues/632#issuecomment-1455109136

essentially the pathway to allowing this in myst-parser, would be to create a plugin in https://github.com/executablebooks/mdit-py-plugins, then we can add it in the list of `enable_extensions`
